app/interactors/file_attachments/preview_interactor.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class FileAttachments::PreviewInteractor
+  include Interactor
+
+  delegate :params,
+           :edition,
+           :attachment_revision,
+           :asset,
+           to: :context
+
+  def call
```

Comment:
Do you mean for this to be in a transaction?

Reply:
At first I thought it wasn't necessary, but the call to 'ensure_assets' could create Asset records, which the PreviewAssetService would then save, so I think we do need it.

Reply:
You're right it can change state so should be within a transaction.